code/game/gamemodes/shadowling/shadowling_abilities.dm

```diff
@@ -0,0 +1,842 @@
+#define EMPOWERED_THRALL_LIMIT 5
+
+/obj/effect/proc_holder/spell/proc/shadowling_check(var/mob/living/carbon/human/H)
+	if(!H || !istype(H)) return
+	if(H.dna.species.id == "shadowling" && is_shadow(H)) return 1
+	if(H.dna.species.id == "l_shadowling" && is_thrall(H)) return 1
+	if(!is_shadow_or_thrall(usr)) to_chat(usr, "<span class='warning'>You can't wrap your head around how to do this.</span>")
+	else if(is_thrall(usr)) to_chat(usr, "<span class='warning'>You aren't powerful enough to do this.</span>")
+	else if(is_shadow(usr)) to_chat(usr, "<span class='warning'>Your telepathic ability is suppressed. Hatch or use Rapid Re-Hatch first.</span>")
+	return 0
+
+
+
+/obj/effect/proc_holder/spell/targeted/glare //Stuns and mutes a human target for 10 seconds
+	name = "Glare"
+	desc = "Stuns and mutes a target for a decent duration."
+	panel = "Shadowling Abilities"
+	charge_max = 300
+	human_req = 1
+	clothes_req = 0
+	action_icon_state = "glare"
+	action_icon = 'icons/mob/actions/actions_sling.dmi'
+
+/obj/effect/proc_holder/spell/targeted/glare/cast(list/targets,mob/user = usr)
+	for(var/mob/living/target in targets)
```

Comment:
this entire ability should be a melee-only targetted spell, like fireball, with the mouse cursor thing.